src/debugger/gdb-stub.c

```diff
@@ -50,7 +50,21 @@ static const char* TARGET_XML = "<target version=\"1.0\">"
                                 "<reg name=\"sp\" bitsize=\"32\" type=\"data_ptr\"/>"
                                 "<reg name=\"lr\" bitsize=\"32\"/>"
                                 "<reg name=\"pc\" bitsize=\"32\" type=\"code_ptr\"/>"
-                                "<reg name=\"cpsr\" bitsize=\"32\" regnum=\"25\"/>"
+                                "<flags id=\"cpsr_flags\" size=\"4\">"
+                                "<field name=\"N\" start=\"31\" end=\"31\"/>"
+                                "<field name=\"Z\" start=\"30\" end=\"30\"/>"
+                                "<field name=\"C\" start=\"29\" end=\"29\"/>"
+                                "<field name=\"V\" start=\"28\" end=\"28\"/>"
+                                "<field name=\"I\" start=\"7\" end=\"7\"/>"
+                                "<field name=\"F\" start=\"6\" end=\"6\"/>"
+                                "<field name=\"T\" start=\"5\" end=\"5\"/>"
+                                "<field name=\"M4\" start=\"4\" end=\"4\"/>"
+                                "<field name=\"M3\" start=\"3\" end=\"3\"/>"
+                                "<field name=\"M2\" start=\"2\" end=\"2\"/>"
+                                "<field name=\"M1\" start=\"1\" end=\"1\"/>"
+                                "<field name=\"M0\" start=\"0\" end=\"0\"/>"
```

Comment:
Why is this split out? The M field functions as a single 5 bit value (or more accurately a single 4 bit value and an always-1 5th bit).

Reply:
Because I read the documentation too literally. I've combined it into one "M" field.